ui/widgets/code_editor.go

```diff
@@ -80,44 +83,21 @@ func (c *CodeEditor) SetOnLoadExample(f func()) {
 }
 
 func (c *CodeEditor) SetCode(code string) {
-	c.editor.SetText(code)
-	c.lines = strings.Split(code, "\n")
+	c.editor.SetText(code, false)
 	c.code = code
 }
 
-func (c *CodeEditor) SetLanguage(_ string) {
+func (c *CodeEditor) SetLanguage(lang string) {
+	c.lang = lang
 }
 
 func (c *CodeEditor) Code() string {
 	return c.editor.Text()
 }
 
 func (c *CodeEditor) Layout(gtx layout.Context, theme *chapartheme.Theme, hint string) layout.Dimensions {
-	for {
-		ev, ok := gtx.Event(
-			key.Filter{
-				Focus: c.editor,
-				Name:  key.NameTab,
-			},
-		)
-		if !ok {
-			break
-		}
-		e, ok := ev.(key.Event)
-		if !ok {
-			continue
-		}
-
-		if e.Name == key.NameTab && e.State == key.Release {
-			c.editor.Insert("    ")
-			break
-		}
-	}
-
 	if ev, ok := c.editor.Update(gtx); ok {
-		if _, ok := ev.(widget.ChangeEvent); ok {
-			c.lines = strings.Split(c.editor.Text(), "\n")
-
+		if _, ok := ev.(giovieweditor.ChangeEvent); ok {
 			if c.onChange != nil {
```

Comment:
It would be better to move call to `c.editor.UpdateTextStyles` here for better performance.

Reply:
Good point